### ISSUE TITLE
Route is not found if declared after declaration of root path route

### ISSUE THREAD

**Issue by techboy-zero** [ReactTraining/react-router#5759]
Hello, I am confused about a behavior of Route. Does order of **Routes** create any impact? Specially does it have anything to do with **Routes** defined before/after defining **Route** for root path?

I am using **react-router** version **2.8.1**.

Code 1:
```
<Router history={history} onUpdate={gaCallback}>
  <Route path="sponsor" component={App} >
    <IndexRoute component={HomePage} />
    <Route path="home" component={HomePage} />
    <Route path="settings" component={SettingsPage} />
    <Route path="*" component={NotFound} status={404} />
  </Route>
  <Route path="/" component={App} >
    <IndexRoute component={HomePage} />
    <Route path="home" component={HomePage} />
    <Route path="settings" component={SettingsPage} />
    <Route path="*" component={NotFound} status={404} />
  </Route>
</Router>
```

Code 2:
```
<Router history={history} onUpdate={gaCallback}>
  <Route path="/" component={App} >
    <IndexRoute component={HomePage} />
    <Route path="home" component={HomePage} />
    <Route path="settings" component={SettingsPage} />
    <Route path="*" component={NotFound} status={404} />
  </Route>
  <Route path="sponsor" component={App} >
    <IndexRoute component={HomePage} />
    <Route path="home" component={HomePage} />
    <Route path="settings" component={SettingsPage} />
    <Route path="*" component={NotFound} status={404} />
  </Route>
</Router>
```

If I write the **Routes** according to the 1st code then everything works fine. But If I change the order to the second one then the later route is not found. For example If I hit **localhost:port/sponsor** the **NotFound** component gets loaded.

So, basically I have a working code but I want to know the reason behind this behavior for avoiding unwanted situation in future.

Thanks in advance.

**Comment by timdorr** [comment]
Yes, order matters. You should move your NotFound route up to the top level and make it come last.
